File: Java/Problem_088.java
import java.util.*;

public class Euler88 {
    static final int KMAX = 12000;
    static final int LIMIT = 2 * KMAX;
    static int[] best = new int[KMAX + 1];

    static void dfs(int start, int prod, int sum, int m) {
        int k = m + (prod - sum);
        if (k <= KMAX) best[k] = Math.min(best[k], prod);

        for (int f = start; f * prod <= LIMIT; f++) {
            dfs(f, prod * f, sum + f, m + 1);
        }
    }

    public static void main(String[] args) {
        Arrays.fill(best, Integer.MAX_VALUE);
        dfs(2, 1, 0, 0);

        HashSet<Integer> set = new HashSet<>();
        for (int k = 2; k <= KMAX; k++) set.add(best[k]);

        long ans = 0;
        for (int x : set) ans += x;

        System.out.println(ans); // 7587457
    }
}

File: Python/Problem_088.py
def euler_88(KMAX=12000):
    LIMIT = 2 * KMAX
    best = [10**18] * (KMAX + 1)

    def dfs(start, prod, s, m):
        k = m + (prod - s)
        if k <= KMAX:
            if prod < best[k]:
                best[k] = prod
        for f in range(start, LIMIT // prod + 1):
            dfs(f, prod * f, s + f, m + 1)

    dfs(2, 1, 0, 0)

    return sum(set(best[2:]))

if __name__ == "__main__":
    print(euler_88())  # 7587457
